File: Tercera Entrega/sdi3-37.Cli-REST/src/com/sdi/presentation/user/MainMenu.java
package com.sdi.presentation.user;

import javax.ws.rs.client.ClientBuilder;
import javax.ws.rs.core.GenericType;
import com.sdi.dto.User;
import com.sdi.presentation.user.action.CrearTareaAction;
import com.sdi.presentation.user.action.ListarCategoriasAction;
import com.sdi.presentation.user.action.MarcarTareaFinalizadaAction;
import com.sdi.presentation.user.action.TareasCategoriaAction;

import alb.util.console.Console;
import alb.util.menu.BaseMenu;

public class MainMenu extends BaseMenu {
	
	public MainMenu() {			
		menuOptions = new Object[][] { 	
			{"USUARIO "+ Sesion.getInstance().getUser().getLogin(), null},
			{ "Listar categorías", 				ListarCategoriasAction.class}, 
			{ "Ver tareas de una categoría", 	TareasCategoriaAction.class},
			{ "Marcar tarea como finalizada", 	MarcarTareaFinalizadaAction.class },
			{ "Crear tarea",					CrearTareaAction.class}
		};
	}

	public static void main(String[] args) {
		String login = Console.readString("Usuario");
		String password = Console.readString("Contraseña");
		
		login(login, password);
		
		if(Sesion.getInstance().getUser() != null) {
			Console.println();
			Console.print("LOGIN CORRECTO, INICIANDO SESIÓN------>");
			new MainMenu().execute();
		}
		else{
			Console.print("\tNo existe el usuario");
		}
	}
	
	private static void login(String login, String password) {
		GenericType<User> modelo = new GenericType<User>() {};
		
		User user = ClientBuilder.newClient()			
			.target(Sesion.getInstance().getRestServiceUrl())
			.path("login")
			.path("username="+login+"&&password="+password)
			.request()
			.get()
			.readEntity(modelo);
		
		Sesion.getInstance().setUser(user);			
	}
}

File: Tercera Entrega/sdi3-37.Cli-REST-C_Sharp/sdi3-37.Cli-REST-C_Sharp/MainMenu.cs
using sdi3_37.Cli_REST_C_Sharp.action;
using System;

namespace sdi3_37.Cli_REST_C_Sharp
{
    class MainMenu : action.Action
    {
        public MainMenu ()
        {

        }

        public void execute()
        {
            //Selecionar opción
            while (true)
            {
                Console.WriteLine("-----------------------------------------------------------------------");
                Console.WriteLine("USUARIO " + Sesion.Instance.User.login);

                //Opción 1: ListarCategoriasAction            
                Console.Write("\t1- Listar categorias\n");

                //Opción 2: TareasCategoriaAction
                Console.Write("\t2- Ver tareas de una categoria\n");

                //Opción 3: MarcarTareaFinalizadaAction
                Console.Write("\t3- Marcar tarea como finalizada\n");

                //Opción 4: CrearTareaAction
                Console.Write("\t4- Crear tarea\n");

                //Opción 0: Salir
                Console.WriteLine();
                Console.Write("\t0 - Salir\n");

                Console.Write("Opción: ");
                switch (Console.ReadLine())
                {
                    case "0":
                        return;
                    case "1":
                        Console.WriteLine();
                        new ListarCategoriasAction().execute();
                        break;
                    case "2":
                        Console.WriteLine();
                        new TareasCategoriaAction().execute();
                        break;
                    case "3":
                        Console.WriteLine();
                        new MarcarTareaFinalizadaAction().execute();
                        break;
                    case "4":
                        Console.WriteLine();
                        new CrearTareaAction().execute();
                        break;
                    default:
                        Console.WriteLine("\tOpción errónea");
                        break;
                }
                Console.WriteLine();
            }
        }       
    }
}
